src/main/java/it/unitn/disi/smatch/data/trees/IBaseNode.java

```diff
@@ -42,6 +42,16 @@
      * @return unmodifiable list of receivers children
      */
     List<E> getChildren();
+    
+    /**
+     * Returns a list of children that can be modified. 
+     * Notice that acting on this list won't fire events.
```

Comment:
This kind of "exceptions" are source of future "gotchas". It's better to keep things uniform.